services/cd-service/pkg/service/batch_test.go

```diff
@@ -0,0 +1,282 @@
+					t.Fatal(err.Error())
+				}
+				// check deployment version
+				{
+					version, err := svc.Repository.State().GetEnvironmentApplicationVersion("production", "test")
+					if err != nil {
+						t.Fatal(err)
+					}
+					if version == nil || *version != 1 {
+						t.Errorf("unexpected version: expected 1, actual: %d", *version)
+					}
+				}
+				// check that the envlock was created
+				{
+					envLocks, err := svc.Repository.State().GetEnvironmentLocks("production")
+					if err != nil {
+						t.Fatal(err)
+					}
+					lock, exists := envLocks["envlock"]
+					if !exists {
+						t.Error("lock was not created")
+					}
+					if lock.Message != "please" {
+						t.Errorf("unexpected lock message: expected \"please\", actual: %q", lock.Message)
+					}
+					_, exists = envLocks["1234"]
+					if exists {
+						t.Error("lock was not deleted")
+					}
+				}
+				// check that the applock was created
+				{
+					appLocks, err := svc.Repository.State().GetEnvironmentApplicationLocks("production", "test")
+					if err != nil {
+						t.Fatal(err)
+					}
+					lock, exists := appLocks["applock"]
+					if !exists {
+						t.Error("lock was not created")
+					}
+					if lock.Message != "please" {
+						t.Errorf("unexpected lock message: expected \"please\", actual: %q", lock.Message)
+					}
+					_, exists = appLocks["5678"]
+					if exists {
+						t.Error("lock was not deleted")
+					}
+				}
+			},
+		},
+		{
+			Name: "more than 100 actions",
+			Setup: []repository.Transformer{
+				&repository.CreateEnvironment{
+					Environment: "production",
+					Config:      config.EnvironmentConfig{Upstream: &config.EnvironmentConfigUpstream{Latest: true}},
+				},
+				&repository.CreateApplicationVersion{
+					Application: "test",
+					Manifests: map[string]string{
+						"production": "manifest",
+					},
+				},
+				&repository.CreateApplicationVersion{
+					Application: "test",
+					Manifests: map[string]string{
+						"production": "manifest2",
+					},
+				},
+			},
+			Test: func(t *testing.T, svc *BatchServer) {
+				ops := []*api.BatchAction {}
+				for i := uint64(1); i <= 100; i++ {	// test exactly 100 first
+					ops = append(ops, &api.BatchAction{
+						Action: &api.BatchAction_Deploy{
+							Deploy: &api.DeployRequest{
+								Environment:  "production",
+								Application:  "test",
+								Version:      2-i%2,  // alternate between 1 and 2
+								LockBehavior: api.LockBehavior_Fail,
+							},
+						},
+					})
+				}
+				_, err := svc.ProcessBatch(
+					context.Background(),
+					&api.BatchRequest{
+						Actions: ops,
+					},
+				)
+				if err != nil {
+					t.Fatal(err)
+				}
+				{
+					version, err := svc.Repository.State().GetEnvironmentApplicationVersion("production", "test")
+					if err != nil {
+						t.Fatal(err)
+					}
+					if version == nil || *version != 2 {
+						t.Errorf("unexpected version: expected 2, actual: %d", *version)
+					}
+				}
+				// add one more action
```

Comment:
This should be a separate test, e.g. have one test with 100 and one test with 101 actions.